src/main/java/igrad/logic/commands/module/ModuleAddAutoCommand.java

```diff
@@ -54,6 +63,32 @@ public CommandResult execute(Model model) throws CommandException {
             throw new CommandException(MESSAGE_DUPLICATE_MODULE);
         }
 
+        for (String prerequisite: prerequisiteModules) {
+            boolean isPrereqPresent = false;
+            ModuleCode prerequisiteModuleCode = new ModuleCode(prerequisite);
+            for (Module module: model.getFilteredModuleList()) {
```

Comment:
consider saving the `model.getFilteredModuleList()` into a var outside this 2 for-loops, else  you keep probing the modules list (overhead incurred)